samples/excel/89-preview-apis/protect-data-in-worksheet-and-workbook-structure.yaml

```diff
@@ -0,0 +1,258 @@
+name: Protect data in worksheet and the workbook structure
+description: Shows how to protect data in a worksheet and the workbook structure.
+host: EXCEL
+api_set:
+    ExcelApi: 1.7
+script:
+    content: |
+        $("#setup").click(() => tryCatch(setup));
+        $("#protect-data-in-worksheet").click(() => tryCatch(protectDataInWorksheet));
+        $("#unprotect-data-in-worksheet").click(() => tryCatch(unprotectDataInWorksheet));
+        $("#protect-workbook-structure").click(() => tryCatch(protectWorkbookStructure));
+        $("#unprotect-workbook-structure").click(() => tryCatch(unprotectWorkbookStructure));
+        $("#password-protect-data-in-worksheet").click(() => tryCatch(passwordProtectDataInWorksheet));
+        $("#password-unprotect-data-in-worksheet").click(() => tryCatch(passwordUnprotectDataInWorksheet));
+        $("#password-protect-workbook-structure").click(() => tryCatch(passwordProtectWorkbookStructure));
+        $("#password-unprotect-workbook-structure").click(() => tryCatch(passwordUnprotectWorkbookStructure));
+
+        async function protectDataInWorksheet() {
+            await Excel.run(async (context) => {
+                await worksheetHelper(context, true, false);
+            });
+        }
+
+        async function unprotectDataInWorksheet() {
+            await Excel.run(async (context) => {
+                await worksheetHelper(context, false, false);
+            });
+        }
+
+        async function protectWorkbookStructure() {
+            await Excel.run(async (context) => {
+                await workbookHelper(context, true, false);
+            });
+        }
+
+        async function unprotectWorkbookStructure() {
+            await Excel.run(async (context) => {
+                await workbookHelper(context, false, false);
+            });
+        }
+
+        async function passwordProtectDataInWorksheet() {
+            await Excel.run(async (context) => {
+                await worksheetHelper(context, true, true, await passwordHandler());
+            });
+        }
+
+        async function passwordUnprotectDataInWorksheet() {
+            await Excel.run(async (context) => {
+                await worksheetHelper(context, false, true, await passwordHandler());
+            });
+        }
+
+        async function passwordProtectWorkbookStructure() {
+            await Excel.run(async (context) => {
+                await workbookHelper(context, true, true, await passwordHandler());
+            });
+        }
+
+        async function passwordUnprotectWorkbookStructure() {
+            await Excel.run(async (context) => {
+                await workbookHelper(context, false, true, await passwordHandler());
+            });
+        }
+
+        async function worksheetHelper(context: Excel.RequestContext, expectProtection: boolean, expectPassword: boolean, password?: string) {
+
+            await passwordHelper(expectPassword, password);
```

Comment:
If you removed `async` from the definition of passwordHelper, you would not need the `await` here. Does passwordHelper really need to be defined as async?